qute.ls/com.redhat.qute.ls/src/main/java/com/redhat/qute/services/QuteCodeActions.java

```diff
@@ -186,15 +197,41 @@ private static void doCodeActionToDisableValidation(Template template, List<Diag
 		codeActions.add(disableValidationForTemplateQuickFix);
 	}
 
+	/**
+	 * Create CodeAction for deprecated `with` Qute syntax.
+	 *
+	 * @param template    the Qute template.
+	 * @param diagnostic  the diagnostic list that this CodeAction will fix.
+	 * @param codeActions the list of CodeActions to perform.
+	 * @throws BadLocationException
+	 *
+	 */
+	private void doCodeActionsForDeprecatedWithSection(Template template, Diagnostic diagnostic,
+			List<CodeAction> codeActions) {
+		String title = MessageFormat.format(QUTE_DEPRICATED_WITH_SECTION, "with");
+		Range withSectionRange = diagnostic.getRange();
+		try {
+			int withSectionStart = template.getTextDocument().offsetAt(withSectionRange.getStart());
+			int withSectionEnd = template.getTextDocument().offsetAt(withSectionRange.getEnd());
+			String replacement = template.getText(withSectionStart, withSectionEnd).replace("#with", "#let")

```

Comment:
I noticed that this would also replace any references to those strings that might occur in the body. I think if a comment or some html might be able to mention it that would be unintended.

Maybe `replaceFirst` for the start element and use `lastIndexOf` with `substring` to replace the ending element.